messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/engine/EngineWebView.kt

```diff
@@ -38,40 +43,60 @@ internal class EngineWebView @JvmOverloads constructor(
         val jsonString = Gson().toJson(configuration)
         encodeToBase64(jsonString)?.let { options ->
             elapsedTimer.start("Engine render for message: ${configuration.messageId}")
-            val messageUrl = "${GistSdk.gistEnvironment.getGistRendererUrl()}/index.html?options=$options"
+            val messageUrl =
+                "${GistSdk.gistEnvironment.getGistRendererUrl()}/index.html?options=$options"
             Log.i(GIST_TAG, "Rendering message with URL: $messageUrl")
-            webView.loadUrl(messageUrl)
-            webView.settings.javaScriptEnabled = true
-            webView.settings.allowFileAccess = true
-            webView.settings.allowContentAccess = true
-            webView.settings.domStorageEnabled = true
-            webView.settings.textZoom = 100
-            webView.setBackgroundColor(Color.TRANSPARENT)
-            webView.addJavascriptInterface(EngineWebViewInterface(this), "appInterface")
-
-            webView.webViewClient = object : WebViewClient() {
-                override fun onPageFinished(view: WebView, url: String?) {
-                    view.loadUrl("javascript:window.parent.postMessage = function(message) {window.appInterface.postMessage(JSON.stringify(message))}")
-                }
-
-                override fun shouldOverrideUrlLoading(view: WebView, url: String): Boolean {
-                    return !url.startsWith("https://code.gist.build")
-                }
-
-                override fun onReceivedError(view: WebView?, errorCod: Int, description: String, failingUrl: String?) {
-                    listener?.error()
-                }
-
-                override fun onReceivedHttpError(view: WebView?, request: WebResourceRequest?, errorResponse: WebResourceResponse?) {
-                    listener?.error()
-                }
-
-                override fun onReceivedError(view: WebView?, request: WebResourceRequest?, error: WebResourceError?) {
-                    listener?.error()
-                }
-
-                override fun onReceivedSslError(view: WebView?, handler: SslErrorHandler?, error: SslError?) {
-                    listener?.error()
+            webView?.let {
```

Comment:
Could be only me, but if the code below doesn't need to execute, I prefer early returns instead of putting inside a block as it looks cleaner to me 😅

Reply:
I am bit hesitant to do an early return because we are logging some important things and running timers that i don't want to miss running.